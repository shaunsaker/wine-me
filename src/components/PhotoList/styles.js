import { StyleSheet } from 'react-native';

import styleConstants from '../../styleConstants';

const ICON_SIZE = 18; // from SectionHeader
const IMAGE_SIZE = 120;

const styles = StyleSheet.create({
  container: {},
  contentContainer: {
    paddingLeft:
      styleConstants.dimensions.spacing.horizontal +
      ICON_SIZE +
      styleConstants.dimensions.spacing.horizontal / 2,
  },
  itemContainer: {
    paddingRight: styleConstants.dimensions.spacing.horizontal,
    paddingBottom: styleConstants.dimensions.spacing.vertical,
  },
  image: {
    width: IMAGE_SIZE,
    height: IMAGE_SIZE,
    borderRadius: styleConstants.dimensions.borderRadius,
    overflow: 'hidden',
  },
});

export default styles;
