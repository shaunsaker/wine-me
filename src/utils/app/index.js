import createError from './createError';
import createMapsLinkFromCoordinates from './createMapsLinkFromCoordinates';
import getDistanceBetweenCoordinates from './getDistanceBetweenCoordinates';
import getGooglePlacesPhotoURI from './getGooglePlacesPhotoURI';
import log from './log';
import navigate from './navigate';
import prepareNextAction from './prepareNextAction';

// All app-specific utils go here
const app = {
  createError,
  createMapsLinkFromCoordinates,
  getDistanceBetweenCoordinates,
  getGooglePlacesPhotoURI,
  log,
  navigate,
  prepareNextAction,
};

export default app;
