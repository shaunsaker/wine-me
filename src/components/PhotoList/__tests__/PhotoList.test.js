import React from 'react';
import renderer from 'react-test-renderer';

import PhotoList from '..';
import PLACE from '../../../mockData/PLACE';
import utils from '../../../utils';

describe('PhotoList', () => {
  const photos = PLACE.photoReferences.map((photoReference) => {
    return {
      uri: utils.app.getGooglePlacesPhotoURI(photoReference),
    };
  });

  describe('renders', () => {
    it('renders with minimum required props', () => {
      const component = renderer.create(<PhotoList data={photos} />);

      expect(component).toMatchSnapshot();
    });
  });

  // NOTE: We don't test render methods (those are handle in the test(s) above)
});
