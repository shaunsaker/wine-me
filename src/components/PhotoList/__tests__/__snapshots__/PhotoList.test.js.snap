// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoList renders renders with minimum required props 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "paddingLeft": 40.5,
    }
  }
  data={
    Array [
      Object {
        "uri": "https://maps.googleapis.com/maps/api/place/photo?maxheight=undefined&photoreference=CmRaAAAAF5bkjtj7bFEGaZqTN1_uZrJMvM6EEOFle3UIDd5JEbZwb281f2F4EH-XR1o6frihMiOzrJEx1KlczmV8o4qLrYygDZE-TKthSRT1SdzvrEQ6AjQbCG6sVeZBR48V0lMLEhDkpyqJh9O9z4Ai-7All7swGhQwTkHxJ77yIZ9d0c8oqWifUrL6eA&key=undefined",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={true}
  initialNumToRender={3}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={Object {}}
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        style={
          Object {
            "paddingBottom": 10,
            "paddingRight": 15,
          }
        }
      >
        <View
          style={Object {}}
        >
          <View
            style={
              Array [
                Array [
                  Object {
                    "flex": 1,
                    "height": 200,
                  },
                  Object {
                    "borderRadius": 5,
                    "height": 120,
                    "overflow": "hidden",
                    "width": 120,
                  },
                ],
                Object {
                  "overflow": "hidden",
                },
              ]
            }
          >
            <FastImageView
              borderRadius={5}
              onFastImageError={[Function]}
              onFastImageLoadEnd={[Function]}
              onFastImageLoadStart={[Function]}
              resizeMode="cover"
              source={
                Object {
                  "uri": "https://maps.googleapis.com/maps/api/place/photo?maxheight=undefined&photoreference=CmRaAAAAF5bkjtj7bFEGaZqTN1_uZrJMvM6EEOFle3UIDd5JEbZwb281f2F4EH-XR1o6frihMiOzrJEx1KlczmV8o4qLrYygDZE-TKthSRT1SdzvrEQ6AjQbCG6sVeZBR48V0lMLEhDkpyqJh9O9z4Ai-7All7swGhQwTkHxJ77yIZ9d0c8oqWifUrL6eA&key=undefined",
                }
              }
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
